Setting a [DEFAULT] configuration in ~/.config/praw.ini makes some tests fail

## Issue Description

With something like the following in `~/.config.praw.ini`:

```
[DEFAULT]
username=myusername
password=mypassword
client_id=myclientid
client_secret=myclientsecret
user_agent=myuseragent
```

I get the following failures when I run `python3 setup.py test`:

```
=================================== FAILURES ===================================
__________________________ TestAuthWeb.test_authorize __________________________

self = <tests.integration.models.test_auth.TestAuthWeb object at 0x10ca024a8>

    def test_authorize(self):
        with self.recorder.use_cassette('TestAuthWeb.test_authorize'):
>           token = self.reddit.auth.authorize(pytest.placeholders.auth_code)

tests/integration/models/test_auth.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <praw.models.auth.Auth object at 0x11012cb70>
code = 'placeholder_auth_code'

    def authorize(self, code):
        """Complete the web authorization flow and return the refresh token.
    
            :param code: The code obtained through the request to the redirect uri.
            :returns: The obtained refresh token, if available, otherwise ``None``.
    
            The session's active authorization will be updated upon success.
    
            """
        authenticator = self._reddit._read_only_core._authorizer._authenticator
        if not isinstance(authenticator, TrustedAuthenticator) or \
           self._reddit.config.username:
>           raise ClientException('authorize can only be used with web apps.')
E           praw.exceptions.ClientException: authorize can only be used with web apps.

praw/models/auth.py:50: ClientException
_________ TestConfig.test_unset_value_has_useful_string_representation _________

self = <tests.unit.test_config.TestConfig object at 0x110647860>

    def test_unset_value_has_useful_string_representation(self):
        config = Config('DEFAULT')
>       assert str(config.password) == 'NotSet'
E       AssertionError: assert 'hunter2' == 'NotSet'
E         - hunter2
E         + NotSet

tests/unit/test_config.py:103: AssertionError
______________ TestReddit.test_reddit__missing_required_settings _______________

self = <tests.unit.test_reddit.TestReddit object at 0x1110f3160>

    def test_reddit__missing_required_settings(self):
        for setting in self.REQUIRED_DUMMY_SETTINGS:
            with pytest.raises(ClientException) as excinfo:
                settings = self.REQUIRED_DUMMY_SETTINGS.copy()
                del settings[setting]
>               Reddit(**settings)
E               Failed: DID NOT RAISE <class 'praw.exceptions.ClientException'>

tests/unit/test_reddit.py:116: Failed
==================== 3 failed, 556 passed in 26.47 seconds =====================
```

After removing the `[DEFAULT]` section, all tests passed as expected. The failures were caused by PRAW finding and using default credentials when the tests expect there to be no such defaults.

I'm not sure if this is something we can fix or work around, or if it would be better to just add a comment in the tests.

## System Information

PRAW Version: 5.4.0.dev0
Python Version: 3.6.1
Operating System: macOS 10.13.3

--- a/tests/integration/models/test_auth.py
+++ b/tests/integration/models/test_auth.py
@@ -19,7 +19,8 @@ class TestAuthWeb(IntegrationTest):
         self.reddit = Reddit(client_id=pytest.placeholders.client_id,
                              client_secret=pytest.placeholders.client_secret,
                              redirect_uri=pytest.placeholders.redirect_uri,
-                             user_agent=pytest.placeholders.user_agent)
+                             user_agent=pytest.placeholders.user_agent,
+                             username=None)
 
     def test_authorize(self):
         with self.recorder.use_cassette('TestAuthWeb.test_authorize'):

--- a/tests/unit/test_config.py
+++ b/tests/unit/test_config.py
@@ -99,5 +99,5 @@ class TestConfig(object):
         assert str(excinfo.value) == 'No short domain specified.'
 
     def test_unset_value_has_useful_string_representation(self):
-        config = Config('DEFAULT')
+        config = Config('DEFAULT', password=Config.CONFIG_NOT_SET)
         assert str(config.password) == 'NotSet'

--- a/tests/unit/test_reddit.py
+++ b/tests/unit/test_reddit.py
@@ -3,6 +3,7 @@ import types
 import mock
 import pytest
 from praw import __version__, Reddit
+from praw.config import Config
 from praw.const import configparser
 from praw.exceptions import ClientException
 from prawcore import Requestor
@@ -112,7 +113,7 @@ class TestReddit(UnitTest):
         for setting in self.REQUIRED_DUMMY_SETTINGS:
             with pytest.raises(ClientException) as excinfo:
                 settings = self.REQUIRED_DUMMY_SETTINGS.copy()
-                del settings[setting]
+                settings[setting] = Config.CONFIG_NOT_SET
                 Reddit(**settings)
             assert str(excinfo.value).startswith('Required configuration '
                                                  'setting \'{}\' missing.'

